db/db_impl/db_impl.h

```diff
@@ -1717,8 +1719,14 @@ class DBImpl : public DB {
 
   Status TrimMemtableHistory(WriteContext* context);
 
+  // Switch memtable without creating new WAL. Also set the next_log_num as
+  // specified in mem_switch_record.
+  //
+  // NOTE: this function should only be used when WAL is disabled
   Status SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context,
-                        std::string replication_sequence);
+                        const MemTableSwitchRecord& mem_switch_record);
```

Comment:
Overload where the only thing that differs is an additional parameter is a bit confusing. Could we change the name of the function?